runtime/monovm-bridge.m

```diff
@@ -419,4 +419,64 @@
 
 #endif // DOTNET
 
+/*
+ * ToggleRef support
+ */
+// #define DEBUG_TOGGLEREF 1
+
+static void
+gc_register_toggleref (MonoObject *obj, id self, bool isCustomType)
+{
+	// COOP: This is an icall, at entry we're in unsafe mode. Managed memory is accessed, so we stay in unsafe mode.
+	MONO_ASSERT_GC_UNSAFE;
+
+#ifdef DEBUG_TOGGLEREF
+	id handle = xamarin_get_nsobject_handle (obj);
+
+	PRINT ("**Registering object %p handle %p RC %d flags: %i isCustomType: %i",
+		obj,
+		handle,
+		(int) (handle ? [handle retainCount] : 0),
+		xamarin_get_nsobject_flags (obj),
+		isCustomType
+		);
+#endif
+	mono_gc_toggleref_add (obj, TRUE);
```

Comment:
What's the difference between `TRUE` here and `true` a few lines further down?

Reply:
None really, it was probably written by different people (this is code that's been moved).